src/ifcopenshell-python/ifcopenshell/type_checker.py

```diff
@@ -0,0 +1,149 @@
+from lark import Lark, Transformer
+
+# Grammar to retrieve the following structure: YYYY-MM-DDThh:mm:ss
+grammar = """
+    start: time_value
+    time_value: year "-" month "-" day "T" hour ":" minute ":" second ["." millisecond]
+
+    hour: DIGIT DIGIT
+    minute: DIGIT DIGIT
+    second: DIGIT DIGIT
+    millisecond: DIGIT+
+
+    day: DIGIT DIGIT
+    month: DIGIT DIGIT
+    year: MINUS? DIGIT+
+
+    MINUS: "-"
+    DIGIT: "0".."9"
+
+    %import common.WS
+    %ignore WS
+"""
+
+parser = Lark(grammar, start='time_value')
+
+
+class DatetimeTransformer(Transformer):
+    def year(self, items):
+        return ''.join(token.value for token in items)
```

Comment:
How about something like this:

```suggestion
    def _integer(self, items):
        return int(''.join(token.value for token in items))

    year = _integer
```

and same as below.

I also added the `int()` I would convert to the appropriate type in the transformer already.

Reply:
changed in https://github.com/IfcOpenShell/IfcOpenShell/pull/3864/commits/5c5aaf16e072d9e96d7c2f99b0cff6945b897f3c